database/factories/BeatmapsetFactory.php

```diff
@@ -57,6 +59,12 @@ public function noDiscussion()
         return $this->state(['discussion_enabled' => false]);
     }
 
+    public function owner(?User $user = null)
+    {
+        return $this->state(['user_id' => $user ?? User::factory()])
+            ->afterMaking(fn (Beatmapset $beatmapset) => $beatmapset->creator = $beatmapset->user->username);
```

Comment:
update `creator` attribute function in definition instead (put at the bottom so `user_id` is set first).

Reply:
`user` isn't part of attributes available in `definition`?

Reply:
it's not but can be looked up `User::find($attr['user_id'])->.. ?? ...`

Reply:
or it can be set without another lookup....

Reply:
Even then this should be part of configuration and not just one specific state (and remove default `creator` in that case).